src/demux/bcl_convert/test.sh

```diff
@@ -44,6 +44,36 @@ echo ">>> Checking whether output fastq files are created"
 [[ ! -f  "$TMPDIR/output2/Undetermined_S0_R1_001.fastq.gz" ]] && echo "Output fastq files could not be found!" && exit 1
 [[ ! -f "$TMPDIR/output2/s1_S1_R1_001.fastq.gz" ]] && echo "Output fastq files could not be found!" && exit 1
 
+echo ">>> Test no lane splitting argument"
+awk '!/NoLaneSplitting/' "$meta_resources_dir/bcl2/sample_sheet.csv" > "$TMPDIR/sample_sheet_without_LaneSplitting.csv"
+
+$meta_executable \
+  --input "$meta_resources_dir/bcl2" \
+  --sample_sheet "$TMPDIR/sample_sheet_without_LaneSplitting.csv" \
+  --output "$TMPDIR/output3" \
+  --test_mode true \
+  --no_lane_splitting true
+
+echo ">>> Checking whether the output dir exists"
+[[ ! -d "$TMPDIR/output3" ]] && echo "Output dir could not be found!" && exit 1
+
+echo ">>> Checking whether output fastq files with lane splitting are created"
+
+[[ ! -f  "$TMPDIR/output3/Undetermined_S0_R1_001.fastq.gz" ]] && echo "Output fastq files could not be found!" && exit 1
+[[ ! -f "$TMPDIR/output3/s1_S1_R1_001.fastq.gz" ]] && echo "Output fastq files could not be found!" && exit 1
+
+$meta_executable \
+  --input "$meta_resources_dir/bcl2" \
+  --sample_sheet "$TMPDIR/sample_sheet_without_LaneSplitting.csv" \
+  --output "$TMPDIR/output4" \
+  --test_mode true \
+  --no_lane_splitting false
+
+echo ">>> Checking whether output fastq files without lane splitting are created"
+
+[[ ! -f  "$TMPDIR/output4/Undetermined_S0_L001_R1_001.fastq.gz" ]] && echo "Output fastq files could not be found!" && exit 1
```

Comment:
Should BCL convert not output the files without the `L001` part because lanes were merged?

Reply:
Weirdly the naming convention is the other way around, 
"Concatenation of FASTQ files separated by lane can be done by enabling this setting. FASTQs will be output with the naming convention
<Sample_ID>_S#_<R or I>#_001.fastq.gz (no L00# included)."
so when set to false, it sticks with the original name including 'L00#'

Reply:
(in the first two tests, the no-lane-splitting is set to true in the sample sheet as well)

Reply:
I see, got it!